### ISSUE TITLE
MAC本地ollama部署2.6版本，‘cblas.h’文件未找到

### ISSUE THREAD

**Issue by zhb-code** [OpenBMB/MiniCPM-V#433]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

1、git clone -b minicpm-v2.6 https://github.com/OpenBMB/ollama.git
     cd ollama/llm
     git clone -b minicpmv-main https://github.com/OpenBMB/llama.cpp.git
     cd ../
2、brew install go cmake gcc
3、go generate ./...
就执行第三步的时候，报错：ollama/llm/llama.cpp/ggml-blas.cpp:12:13: fatal error: 'cblas.h' file not found
我已经安装了openblas，但是一直报这个错，无论怎么处理都不行，一直报这个错。

### 期望行为 | Expected Behavior

可以正常执行。

### 复现方法 | Steps To Reproduce

1、git clone -b minicpm-v2.6 https://github.com/OpenBMB/ollama.git
     cd ollama/llm
     git clone -b minicpmv-main https://github.com/OpenBMB/llama.cpp.git
     cd ../
2、brew install go cmake gcc
3、go generate ./...

### 运行环境 | Environment

```Markdown
- OS:macOS 14.5
- Python:3.11
- Transformers:
- PyTorch:
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):
```

### 备注 | Anything else?
您可以创建一个干净的环境，按照文档步骤，拉取代码，安装cmake，然后执行go generate ./...  就会报这个错
_No response_

**Comment by zhb-code** [comment]
我自己用brew安装了openblas，还是报错

**Comment by BothSavage** [comment]
全局搜索下这个文件，复制这个文件夹路径。然后在llm/llama.cpp/CMakeLists.txt中添加

# Include directories
include_directories(/dir_path)

**Comment by zhb-code** [comment]
我按照你的方法，找到cblas.h文件，路径是：/opt/homebrew/Cellar/openblas/0.3.27/include，
然后在llm/llama.cpp/CMakeLists.txt中添加如下：
![1723195236547](https://github.com/user-attachments/assets/f8dcd9e0-8abd-491b-b16c-e7de7318f211)

然后执行go generate ./...   还是报那个错，CMakeLists.txt中刚添加的两句话也没了，CMakeLists.txt这个文件是不是受其他文件影响会自动修改？

**Comment by bruiselee** [comment]
同情况。可能是fork的ollama版本的构建脚本问题，因为同环境下，单独编译minicpm-main分支的llama.cpp是成功的（make），单独clone 官方ollama（最新/连带submodule的llama.cpp）也是构建成功的(go generate与go build)

**Comment by BothSavage** [comment]
<img width="1225" alt="image" src="https://github.com/user-attachments/assets/fada4aa1-53d4-4668-bcb0-17ac55c18928">
我这样是可以写后go generate ./...没有问题了
但是我下一步go build . 报错了

**Comment by WangchenhaoHZ** [comment]
苹果准备添加自己的blas，所以拒绝了第三方的。感觉要更新到macos15试试。
<img width="928" alt="image" src="https://github.com/user-attachments/assets/3f3a82c5-5e45-49d8-a53d-bc92ffbadf98">

**Comment by BothSavage** [comment]
MACOS版本：版本15.0 Beta版(24A5279h) || 版本15.1 Beta版(24B5009l)
git版本：58a14c37

1. go generate ./...问题的解决就是在CMakeLists.txt中include对应的文件就OK
2. go build . 问题报错，export CGO_LDFLAGS="-framework Accelerate"这样就行
3.最后运行ollama版本的时候出现 llama_get_logits_ith: invalid logits id 10, reason: no logits
![image](https://github.com/user-attachments/assets/801607eb-5855-489f-9fd4-0d284a477a66)

#432

**Comment by zhb-code** [comment]
还是不行，CMakeLists.txt中include对应的文件，一执行go generate ./...后，CMakeLists.txt刚加的那句include就自动没了，然后还是报这个错

**Comment by BothSavage** [comment]
> 还是不行，CMakeLists.txt中include对应的文件，一执行go generate ./...后，CMakeLists.txt刚加的那句include就自动没了，然后还是报这个错

奇了怪了，你试试webdemo吧，ollama最终都会报错invalid logits id 10, reason: no logits

**Comment by zhb-code** [comment]
是的，最终都会报错invalid logits id 10, reason: no logits，因为我一个月之前曾经跑起来2.5，后来更新代码之后，聊天就一直报这个，我看到bug链接了，只能等他们解决了

**Comment by zhb-code** [comment]
我当时就是因为报invalid logits id 10, reason: no logits这个错，才删掉之前编译好的，想重新拉取代码从头编译一次，结果就遇到了‘cblas.h’文件未找到这个问题，然后一直搞不定

**Comment by zhb-code** [comment]
> MACOS版本：版本15.0 Beta版(24A5279h) || 版本15.1 Beta版(24B5009l) git版本：58a14c37
> 
> 1. go generate ./...问题的解决就是在CMakeLists.txt中include对应的文件就OK
> 2. go build . 问题报错，export CGO_LDFLAGS="-framework Accelerate"这样就行
>    3.最后运行ollama版本的时候出现 llama_get_logits_ith: invalid logits id 10, reason: no logits
>    ![image](https://private-user-images.githubusercontent.com/38107458/356682888-801607eb-5855-489f-9fd4-0d284a477a66.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM1MjExNDcsIm5iZiI6MTcyMzUyMDg0NywicGF0aCI6Ii8zODEwNzQ1OC8zNTY2ODI4ODgtODAxNjA3ZWItNTg1NS00ODlmLTlmZDQtMGQyODRhNDc3YTY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODEzVDAzNDcyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliM2VkZGZiMDdmNGZlODNkYTdhYWFjZjNiNTM3MzZmYTMyMWYxMTU4YTcxNTlhNjFiNGMyZWI3N2NlYTIwNTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VrQivy-blRnNpCERZbYZt3GAIF2bhwrV9yVR5A_m5gQ)
> 
> #432

我今天重新拉取了代码，按照你的方法，go generate ./...这一步已经通过了，go build .这一步报错了，报错信息为：
/opt/homebrew/Cellar/go/1.22.6/libexec/pkg/tool/darwin_arm64/link: running cc failed: exit status 1
Undefined symbols for architecture arm64:
  "_openblas_set_num_threads", referenced from:
      ggml_backend_blas_graph_compute(ggml_backend*, ggml_cgraph*) in libllama.a[9](ggml-blas.cpp.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
我执行了export CGO_LDFLAGS="-framework Accelerate"这个也不行。

**Comment by zhb-code** [comment]
感谢 @BothSavage @HuPengCheng,通过你们的方法解决了这个问题，目前就剩invalid logits id 10, reason: no logits这个了

**Comment by williamlee1982** [comment]
> 感谢 @BothSavage @HuPengCheng,通过你们的方法解决了这个问题，目前就剩invalid logits id 10, reason: no logits这个了

老兄，最后的 invalid logits id 10这个问题解决了没？
